packages/eslint-plugin/src/rules/ban-instanceof-html-element/index.js

```diff
@@ -0,0 +1,115 @@
+// @ts-check
+const { AST_NODE_TYPES } = require('@typescript-eslint/experimental-utils');
+const createRule = require('../../utils/createRule');
+
+/**
+ * @typedef {import("@typescript-eslint/types/dist/ts-estree").BinaryExpression} BinaryExpression
+ * @typedef {import("@fluentui/react-utilities/src/utils/isHTMLElement").HTMLElementConstructorName} HTMLElementConstructorName
+ *
+ */
+
+module.exports = createRule({
+  name: 'ban-instanceof-html-element',
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Ban usage of instanceof HTMLElement comparison',
+      category: 'Possible Errors',
+      recommended: 'error',
+    },
+    messages: {
+      invalidBinaryExpression: 'instanceof {{right}} should be avoided, use isHTMLElement instead.',
```

Comment:
please add documentation of the rule to the README https://github.com/microsoft/fluentui/blob/565c7fa43a01190322ebb090b14ec89cc63d95f8/packages/eslint-plugin/README.md